controller/update.go

```diff
@@ -268,15 +266,13 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
 			}
 			if ok {
 				go func() {
-					v := atomic.AddInt64(&removeInProgress, 1)
-					removeInProgress = v
+					atomic.AddInt64(&removeInProgress, 1)
 					err := c.removeFirst(ctx, newReq, opts)
 					if err != nil {
 						fail <- maskAny(err)
 						return
 					}
-					v = atomic.AddInt64(&removeInProgress, -1)
-					removeInProgress = v
+					atomic.AddInt64(&removeInProgress, -1)
```

Comment:
Makes sense. Anyway I am really confused about this case and I looked that up what to do about it.